fcrepo-client-impl/src/main/java/org/fcrepo/client/impl/FedoraDatastreamImpl.java

```diff
@@ -93,9 +99,61 @@ public void setGraph( final Graph graph ) {
         hasContent = getTriple( subject, DESCRIBES ) != null;
     }
 
+    /**
+     * Retrieves the URI path for the properties of a Datastream resource
+     * Inspects the headers in order to determine if another resource describes this resource
+     * @see FedoraRepositoryImpl#exists
+     * @author griffinj@lafayette.edu
+     *
+     */
     @Override
     public String getPropertiesPath() {
-        return path + "/" + FedoraJcrTypes.FCR_METADATA;
+
+        String descriptionPath = "/" + path + "/" + FedoraJcrTypes.FCR_METADATA;
+
+        final HttpHead head = httpHelper.createHeadMethod(path);
+
+        try {
+            final HttpResponse response = httpHelper.execute(head);
+            final StatusLine status = response.getStatusLine();
+            final int statusCode = status.getStatusCode();
+            final String uri = head.getURI().toString();
+            if (statusCode == SC_OK) {
+
+                final Header[] linkHeaders = response.getHeaders("Link");
+
+                if (linkHeaders != null) {
+                    final List<Header> headers = Arrays.asList(linkHeaders);
+                    final List<Link> descLinks = headers.stream()
+                                                        .map(u -> Link.valueOf(u.getValue()))
+                                                        .filter(v -> v.getRel().equals("describedby"))
+                                                        .collect(Collectors.toList());
+                    if (!descLinks.isEmpty()) {
+
+                        final Link descLink = descLinks.stream().findFirst().get();
+
+                        // This must be a relative URL
+                        descriptionPath = descLink.getUri().toString().replace(repository.getRepositoryUrl(), "");
```

Comment:
Caching the `descriptionPath` would likely make sense since its value can be assumed not to be dynamic between client requests.